openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -1535,6 +1539,31 @@ def test_fetching_program_discounted_price(self):
         )
         self.assertEqual(data['discount_data'], mock_discount_data)
 
+    @httpretty.activate
+    @override_switch("{}.{}".format(
+        PROGRAMS_UTILS_WAFFLE_SWITCH_NAMESPACE.name,
+        ALWAYS_CALCULATE_PROGRAM_PRICE_AS_ANONYMOUS_USER.switch_name
```

Comment:
Yuck.  Is this really how we have to do this?  There isn't a way to get the namespaced name off the switch?

Reply:
Apparently there was for WaffleFlags but not for a switch https://github.com/edx/edx-platform/pull/18390